src/com/dbconnectors/DbConnector.java

```diff
@@ -0,0 +1,51 @@
+package com.dbconnectors;
+
+import java.sql.*;
+
+public class DbConnector extends Thread {
```

Comment:
This is completely wrong, the idea of the PoC is not for the Db operation to be a thread, it is a simple insert operation, you are not going to gain anything by making it a different thread. The thread should be used to search the file, thus if the file is very large, your application will not hang while the search is happening.

Reply:
changed the thread to SearchWord class